scheduler/src/cook/mesos/api.clj

```diff
@@ -57,69 +83,167 @@
+                  [])
+        mkparams (fn [cid params]
+                   (let [param-id (d/tempid :db.part/user)]
+                         (if (empty? params) {}
+                             {:docker/parameters
+                              (mapv (fn [kvp]
+                                      (let
+                                          [k (:key kvp)
+                                           v (:value kvp)
+                                           ]
+                                        {:docker.param/key k
+                                         :docker.param/value v}
+                                        )) params)})))
+        mkvolumes (fn [cid vols]
+                    (let [modemap { "RO" 2 "RW" 1 }]
+                      (if (empty? vols) {}
+                          {:container/volumes
+                           (mapv
+                            (fn [m]
+                              (reduce
+                               (fn [b kvp]
+                                 (into b
+                                       (cond
+                                         (= (key kvp) :container_path)
+                                         {:container.volume/container_path (val kvp)}
+                                         (= (key kvp) :host_path)
+                                         {:container.volume/host_path (val kvp)}
+                                         (= (key kvp) :mode)
+                                         {:container.volume/mode ;;(get modemap (clojure.string/upper-case (val kvp)))}
+                                          (val kvp)}
+                                         ))) {} m))vols)})))]
+    (cond
+      (= (clojure.string/lower-case ctype) "docker")
+      (let [docker (:docker container)
+            params (if (contains? docker :parameters)
+                     (:parameters docker)
+                     [])
+            netmap { "HOST" 1 "BRIDGE" 2 "NONE" 3}]
+        [[:db/add id :job/container container-id]
+         (into {:db/id container-id
+                :container/type "DOCKER"}
+               ;;:container/type 1 }
+               (mkvolumes container-id volumes))
+         [:db/add container-id :container/docker docker-id]
+         (into {:db/id docker-id
+                :docker/image (:image docker)
+                :docker/network ;;(get netmap
+                                ;;     (clojure.string/upper-case
+                                ;;      ))}
+                (:network docker)}
+               (mkparams docker-id params))
+               ])
+      :else
+      {})))
+
 (sm/defn submit-jobs
   [conn jobs :- [Job]]
-  (doseq [{:keys [uuid command max-retries max-runtime priority cpus mem user name ports uris env]} jobs
+  (doseq [{:keys [uuid command max-retries max-runtime priority cpus mem user name ports uris env container]} jobs
           :let [id (d/tempid :db.part/user)
-                ports (mapv (fn [port]
```

Comment:
Can you return this indent as it was so that I can see th e diff better? Thanks!